assets/model_monitoring/components/tests/unit/test_model_monitor_data_joiner.py

```diff
@@ -0,0 +1,143 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""This file contains unit tests for the Model Monitor Data Joiner component."""
+
+import pytest
+
+from pyspark.sql.types import StructType
+from src.model_monitor_data_joiner.run import join_data
+from tests.e2e.utils.io_utils import create_pyspark_dataframe
+
+LEFT_JOIN_COLUMN = 'left_join_column'
+RIGHT_JOIN_COLUMN = 'right_join_column'
+test_data = [('False', 'True'), ('True', 'False')]
+
+
+def _generate_left_data_df(contains_join_column):
+    left_data = [
+        (3, 4, 1),
+        (6, 7, 2),
+        (9, 10, 3)
+    ]
+    left_columns = ['sepal_length', 'petal_length']
+
+    if contains_join_column is True:
+        left_columns.append(LEFT_JOIN_COLUMN)
+    else:
+        left_columns.append('random_column')
+
+    return create_pyspark_dataframe(left_data, left_columns)
+
+
+def _generate_right_data_df(contains_join_column):
+    right_data = [
+        ('flower1', 1),
+        ('flower2', 2),
+        ('flower3', 3)
+    ]
+    right_columns = ['target']
+
+    if contains_join_column is True:
+        right_columns.append(RIGHT_JOIN_COLUMN)
+    else:
+        right_columns.append('random_column')
+    return create_pyspark_dataframe(right_data, right_columns)
+
+
+@pytest.mark.unit
+class TestModelMonitorDataJoiner:
+    """Test class for model monitor data joiner component."""
+
+    def test_join_data_of_equal_length_successful(self):
+        """Test join data to produce result successfully."""
+        left_data_df = _generate_left_data_df(True)
+        right_data_df = _generate_right_data_df(True)
+
+        joined_data_df = join_data(
+            left_data_df,
+            'left_data',
+            LEFT_JOIN_COLUMN,
+            right_data_df,
+            'right_data',
+            RIGHT_JOIN_COLUMN
+        )
+
+        assert len(joined_data_df.columns) == 5
+        assert joined_data_df.count() == 3
+
+    @pytest.mark.parametrize("left_data_has_join_column, right_data_has_join_column", test_data)  # noqa
+    def test_join_data_missing_join_column_raises_exception(
+        self,
+        left_data_has_join_column,
+        right_data_has_join_column
+    ):
+        """Test join data to raises exception due to missing join column."""
+        # Left does not contain join column
+        left_data_df = _generate_left_data_df(left_data_has_join_column)
+        right_data_df = _generate_right_data_df(right_data_has_join_column)
+
+        with pytest.raises(Exception):
+            join_data(
+                left_data_df,
+                'left_data',
```

Comment:
looks like join_data takes only 4 args - is 'left_data'/'right_data' extra arguments in this case?

Reply:
Thank you